stdlib/universe/filter.go

```diff
@@ -170,3 +173,32 @@ func (t *filterTransformation) UpdateProcessingTime(id execute.DatasetID, pt exe
 func (t *filterTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
+
+// MergeFilterAnyRule removes Filter nodes whose predicate always evaluates to true.
+type MergeFilterAnyRule struct{}
+
+func (MergeFilterAnyRule) Name() string {
+	return "MergeFilterAnyRule"
+}
+
+func (MergeFilterAnyRule) Pattern() plan.Pattern {
+	return plan.Pat(FilterKind, plan.Any())
+}
+
+func (MergeFilterAnyRule) Rewrite(filterNode plan.PlanNode) (plan.PlanNode, bool, error) {
+	if len(filterNode.Predecessors()) != 1 {
```

Comment:
You have just one predecessor pattern (`plan.Any()`), so if there were not exactly one predecessor the pattern would not match.  So this check is already enforced by the pattern.

Reply:
Great I wasn't sure about that so I figured I'd be safe, but I'll update this and the docs.

Reply:
After a reread of the docs its clear now. I was thinking that more parameters to the Pat function went further up the tree but now I see that it just means more predecessors period. The docs are clear I just had to read them twice.